cst438-project-airline/src/main/java/cst438/service/AirlineService.java

```diff
@@ -74,27 +74,53 @@ public AirlineService(FlightRepository flightRepository,
     return flightRepository.findAll();
   }
 
+  /**
+   * Creates a new reservation using the flight ID, user ID, seat ID, first name and last name
+   * specified. Returns the newly created reservation object or null if any of the IDs entered were
+   * incorrect.
+   */
   public Reservation makeReservation(int flightId, int userId, int seatId,
       String passengerFirstName, String passengerLastName) {
-    ArrayList<User> user = userRepository.findByUserId(userId);
-    ArrayList<Flight> flight = flightRepository.findByFlightId(flightId);
-    ArrayList<Seat> seat = seatRepository.findBySeatId(seatId);
+    User user = userRepository.findByUserId(userId);
+    Flight flight = flightRepository.findByFlightId(flightId);
+    Seat seat = seatRepository.findBySeatId(seatId);
 
-    if (user.isEmpty() || flight.isEmpty() || seat.isEmpty()) {
+    // Check if any of the entered IDs are invalid
+    if (user == null || flight == null || seat == null) {
       return null;
     }
 
-    // TODO: Change seat to no longer available
+    // Set seat to unavailable
+    seatRepository.setSeatToUnavailable(seatId);
 
-    // TODO: Make date created time stamp work correctly
+    Reservation reservation = new Reservation(user, passengerFirstName, passengerLastName, flight,
+        seat, flight.getPrice());
 
-    Reservation reservation = new Reservation(user.get(0), passengerFirstName, passengerLastName,
-        flight.get(0), seat.get(0), flight.get(0).getPrice());
     reservationRepository.save(reservation);
 
     return reservation;
```

Comment:
Should we call the db to get the reservation obj instead of returning the one that is created here? I think that would be better proof that the reservation was actually made, and it would also include the timestamp. What do you think?

Reply:
Yes, I was doing it that way before and changed it because I thought it was cleaner, but its causing issues, so i will change it back.